### PR TITLE
add package signing on publish

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
part of #75 

TODO:
- [x] split up public key in to tag and base64
- [x] allow importing of public keys manually
- [x] upload the signed result to the server
- [x] Verify on install

### CODE REVIEW

File: graphql/queries/get_package_version_manifest.graphql [hunk 1]
```diff
@@ -7,5 +7,6 @@ query GetPackageVersionManifestQuery ($name: String!, $version: String!) {
             name
         }
         manifest
+        signature
```
Comment (syrusakbary) [inline graphql/queries/get_package_version_manifest.graphql]: I think you can delete this file.

File: src/commands/keys.rs [hunk 2]
```diff
@@ -83,34 +106,52 @@ pub fn keys(options: KeyOpt) -> Result<(), failure::Error> {
...
+            });
+            let response_or_err: Result<publish_public_key_mutation::ResponseData, _> =
+                graphql::execute_query(&q);
+            match response_or_err {
+                Ok(_) => println!("Key pair successfully added!"),
+                Err(e) => {
+                    error!("Failed to upload public key to server: {}", e);
+                    #[cfg(feature = "telemetry")]
+                    sentry::integrations::failure::capture_error(&e);
+                    info!("Rolling back database update")
```
Comment (syrusakbary) [inline src/commands/keys.rs]: How are we rolling back the update?

Comment (MarkMcCaskey) [reply src/commands/keys.rs]: Good catch, currently it's not

File: end-to-end-tests/verification.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+#alias wapm=target/debug/wapm
+wapm config set registry.url "https://registry.wapm.dev"
```
Comment (syrusakbary) [inline end-to-end-tests/verification.sh]: Love this ❤️ (it help us not pollute the prod one)

**Comment by MarkMcCaskey** [comment]
Great!  Merging now.  There's follow up work to do, especially around testing and and the UX for edge cases

**Pull request merged**